### PR TITLE
[2/2] Support SSL encryption

### PR DESCRIPTION
**Pull request by adamstegman**
https://www.pivotaltracker.com/story/show/111253038

Set up Shield with the necessary certificate information to encrypt traffic inside and outside the Elasticsearch cluster.

Detailed instructions are in the README—basically the suggestion is to use a self-signed CA so you can sign certificates with a SAN with private IP addresses. Based on [the Elastic documentation](https://www.elastic.co/guide/en/shield/current/ssl-tls.html).

### CODE REVIEW

**Comment by jeremiemv** [comment]
Subject Alternative Name... :shipit:

**Comment by adamstegman** [comment]
Hmm, looks like this needs a touchup to address updates. :(

**Comment by adamstegman** [comment]
Reopened with some additional changes to support updates.

* the [default drain script](https://github.com/logsearch/logsearch-boshrelease/blob/v201.0.0/jobs/elasticsearch/templates/bin/drain.erb) for elasticsearch does not support SSL *or* authentication, so I had to replace it
* elasticsearch configuration could be blown away by the updates to elasticsearch. I reordered the configuration steps to ensure this job wins.
* detection of elasticsearch restarts was often wrong since it assumed `http`—I'm using `lsof` instead now to avoid that issue.

File: spec/support/bosh.rb [hunk 1]
```diff
@@ -66,10 +71,19 @@ def bosh_cli(args, allow_failure: false, log: true)
     end
 
     result = ""
-    Dir.chdir(File.expand_path('../../..', __FILE__)) do
-      result = `bosh -n -t #{target} -d #{deployment} #{args}`
-      $stderr.puts result
+    SpinningCursor.run do
```
Comment (jeremiemv) [inline spec/support/bosh.rb]: Cool

File: Gemfile [hunk 2]
```diff
@@ -1,4 +1,5 @@
 source 'https://rubygems.org'
 
-gem 'bosh_cli'
+gem 'bosh_cli', '1.3202.0'
```
Comment (mitch-lindsay) [inline Gemfile]: Have they really had 3000+ minor releases?

**Comment by mitch-lindsay** [comment]
:shipit:

File: Gemfile (hunk 2 above)
Comment (adamstegman) [inline Gemfile]: Super semantic versioning.

It's actually the Jenkins build number :older_man:

**Pull request merged**